### PR TITLE
Team 5: Added JSON saving and loading functionality

### LINKED ISSUE #15: Team 5: Saving
# **Description**
**Task:** Implement Saves.
**Feature:** Team 5: Quests and Progression(#13)

A saving/loading system will be implemented into the game. This will be a module that implements behaviour for multiple data structures to be both saved at certain points in the game, as well as loaded at runtime.
 
# **Milestones**
List of steps that need to be completed for this task.
- [x] Create class structure for generic implementation
- [x] Implement writing to JSON for saves
- [x] Implement wiki for saving and distribute to relevant teams
 
**Completion Deadline:** Aug. 30
 
# **Documentation**
- [Save Handler Documentation](../wiki/Game-State-Saves)
 
# **Member**
- Lachlan (@lellis0406)

### PR DESCRIPTION
**Pull request by lellis0406**
Uses JSON to save and load game state/achievements. See wiki (https://github.com/UQcsse3200/2024-studio-2/wiki/Game-State-Saves) for functionality details.

Closes #15 .

### CODE REVIEW

**Review by LucasRyall — APPROVED** [review]
Functionally solid, small change with adding doc string to deleteJson().

File: source/core/src/main/com/csse3200/game/files/FileLoader.java [hunk 1]
```diff
@@ -87,6 +87,17 @@ public static void writeClass(Object object, String filename, Location location)
     file.writeString(json.prettyPrint(object), false);
   }
 
+  public static void deleteJson(String filename, Location location) {
```
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/files/FileLoader.java:90]: Please add a doc string here.

File: .gitignore [hunk 2]
```diff
@@ -2,5 +2,6 @@
 
 ### Gradle ###
 .gradle/
+*.json
```
Comment (mitsudz) [inline .gitignore:5]: Do not add .json to gitignore - a number of asset configurations are stored in .json files and may need to be modified (find another way to ignore loading/saving files)

File: source/core/src/main/com/csse3200/game/files/FileLoader.java (hunk 1 above)
Comment (BaileyS03) [inline source/core/src/main/com/csse3200/game/files/FileLoader.java:90]: Please add a doc string here

**Review by BaileyS03 — CHANGES_REQUESTED** [review]
Code looks good, and there is good coverage. Please add the doc string where I added a comment.

**Review by mitsudz — COMMENTED** [review]
One change requested - all looks good (not too much significant code changed since the last time I looked at this - mostly just rearranging and integrating into current codebase it looks like).

**Review by mitsudz — APPROVED** [review]

**Review by BaileyS03 — APPROVED** [review]
Looks good thanks for fixing the doc strings!

**Pull request merged**